app/db/repository/authenticator_repository.rb

```diff
@@ -0,0 +1,68 @@
+module DB
+  module Repository
+    class AuthenticatorRepository
+      def initialize(data_object:, resource_repository: ::Resource, logger: Rails.logger)
+        @resource_repository = resource_repository
+        @data_object = data_object
+        @logger = logger
+      end
+
+      def find_all(type:, account:)
+        @resource_repository.where(
+          Sequel.like(
+            :resource_id,
+            "#{account}:webservice:conjur/#{type}/%"
+          )
+        ).all.map do |webservice|
+          load_authenticator(account: account, id: webservice.id.split(':').last, type: type)
+        end.compact
+      end
+
+      def find(type:, account:,  service_id:)
+        webservice =  @resource_repository.where(
+          Sequel.like(
+            :resource_id,
+            "#{account}:webservice:conjur/#{type}/#{service_id}%"
```

Comment:
It looks like this function is trying to find a specific webservice but wouldn't the `%` at the end allow it to pickup other rows from the db we don't necessarily want? Or am I missing something with this function?

Reply:
good catch @telday!